src/cloudsc_hip/CMakeLists.txt

```diff
@@ -0,0 +1,168 @@
+    )
+
+    target_include_directories(dwarf-cloudsc-hip-hoist-lib PUBLIC $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>)
+    target_link_libraries(dwarf-cloudsc-hip-hoist-lib PUBLIC hip::device Serialbox::Serialbox_C $<${HAVE_OMP}:OpenMP::OpenMP_C>)
+
+    if (NOT DEFINED CMAKE_HIP_ARCHITECTURES)
+      message(WARNING "No HIP architecture is set! ('CMAKE_HIP_ARCHITECTURES' is not defined)")
+    else()
+      target_compile_options(dwarf-cloudsc-hip-hoist-lib PRIVATE --offload-arch=${CMAKE_HIP_ARCHITECTURES})
+    endif()
+
+    ecbuild_add_executable(
+        TARGET  dwarf-cloudsc-hip-hoist
+        SOURCES dwarf_cloudsc.cpp
+        LIBS dwarf-cloudsc-hip-hoist-lib
+    )
+
+    ## 
+
+    ###### SCC-K-CACHING-HIP ####
+    ecbuild_add_library(
+        TARGET dwarf-cloudsc-hip-k-caching-lib
+    	INSTALL_HEADERS LISTED
+        SOURCES
+    	    cloudsc/yoecldp_c.h
+    	    cloudsc/load_state.h
+            cloudsc/load_state.cpp
+            cloudsc/cloudsc_c_k_caching.h
+            cloudsc/cloudsc_c_k_caching.cpp
+            cloudsc/cloudsc_driver.h
+            cloudsc/cloudsc_driver.cpp
+            cloudsc/cloudsc_validate.h
+            cloudsc/cloudsc_validate.cpp
+            cloudsc/mycpu.h
+            cloudsc/mycpu.cpp
+        PUBLIC_INCLUDES
+            $<INSTALL_INTERFACE:include>
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>
+        PUBLIC_LIBS
+    	    hip::device
+            Serialbox::Serialbox_C
+            $<${HAVE_OMP}:OpenMP::OpenMP_C>
+    )
+
+    target_include_directories(dwarf-cloudsc-hip-k-caching-lib PUBLIC $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>)
+    target_link_libraries(dwarf-cloudsc-hip-k-caching-lib PUBLIC hip::device Serialbox::Serialbox_C $<${HAVE_OMP}:OpenMP::OpenMP_C>)
+
+    if (NOT DEFINED CMAKE_HIP_ARCHITECTURES)
+      message(WARNING "No HIP architecture is set! ('CMAKE_HIP_ARCHITECTURES' is not defined)")
+    else()
+      target_compile_options(dwarf-cloudsc-hip-k-caching-lib PRIVATE --offload-arch=${CMAKE_HIP_ARCHITECTURES})
+    endif()
+
+    ecbuild_add_executable(
+        TARGET  dwarf-cloudsc-hip-k-caching
+        SOURCES dwarf_cloudsc.cpp
+        LIBS dwarf-cloudsc-hip-k-caching-lib
+    )
+
+    ## 
+
+    
+    
+    # Create symlink for the input data
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
+        ${CMAKE_CURRENT_SOURCE_DIR}/../../data ${CMAKE_CURRENT_BINARY_DIR}/../../../data )
+
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-hip-serial
+        COMMAND bin/dwarf-cloudsc-hip
+        ARGS 1 100 16
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
+        OMP 1
+    )
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-hip-omp
+        COMMAND bin/dwarf-cloudsc-hip
+        ARGS 4 100 16
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
+        OMP 4
+        CONDITION HAVE_OMP
+    )
```

Comment:
Sorry, should have checked this earlier. This one is redundant as we don't have multi-threading in the GPU variants, but we should instead add tests for the hoist and k-caching variants as well.